lib/fs.js

```diff
@@ -1639,39 +1649,45 @@ fs.realpathSync = function realpathSync(p, options) {
     pos = nextPartRe.lastIndex;
 
     // continue if not a symlink
-    if (knownHard[base]) {
+    if (knownHard[base] || (cache && cache.get(base) === base)) {
       continue;
     }
 
     var resolvedLink;
-    var stat = fs.lstatSync(base);
-    if (!stat.isSymbolicLink()) {
-      knownHard[base] = true;
-      continue;
-    }
+    if (cache && cache.has(base)) {
```

Comment:
Would it be faster to just do `cache.get()` and check if it returned `undefined`?